### DISCUSSION TITLE
isn't a terminal gui really a tui? why not rename?

### DISCUSSION THREAD

**Discussion by cnervi** [tui-cs/Terminal.Gui#2052] [Ideas]
maybe tui.net or something like this, just saying, but really goo project, it touches all my buttons

**Comment by tznind** [comment]
Well yes tui is more accurate, but its a bit late in the day for a rebranding!  Also I would say that GUI is a more familiar abreviation to most programmers than TUI and therefore helps with discoverability of the library.  Especially for those who are comming from an GUI/Mobile background and want to create "a GUI that works on terminals".  

Searching for TUI still shows the package in NuGet search so it's clearly still discoverable that way too.

I'm glad that you are enjoying the library :)

**Reply by cnervi** [reply]
You still can make transitional packages an call it Terminal.UI as a more neutral name, don't take the wrong way I'm jus offering ideas, not pushing. Still haven't used it but seems like a treat, will definitely the next thing in my list, I'm constantly battling to choose a form of UI and always end up with wxWidgets or GTK also since a child terminal interface like the one that was used in supermarket and such seems just so cool to me, even before I knew how to program, cheers, will be lurking and trying when I have the time

**Reply by tznind** [reply]
That's great, let us know when you do try it, new user feedback is always welcome.  Just incase you hadn't seen it there is also a visual designer for the library that may make getting into it easier: https://github.com/gui-cs/TerminalGuiDesigner .

**Reply by cnervi** [reply]
nice, first I need to get my C# game on par, been years, but this maked me really intrested in coming back surely will try the designer as your lib, and keep bothering you ahahah, cheers, ty for bring new live to the therminals

**Reply by cnervi** [reply]
duuude the designer is better than som gui designers that I have used, didn't read the docs just dl an run it and knew how to to basic things, congrats

**Reply by tznind** [reply]
Thanks! both for taking the time to try it out and to feedback how it went! So glad to hear that it was a good experience.  Welcome back to C# :)

**Comment by BDisp** [comment]
Imagine we rename it to TUI (Text User Interface) and later it can display images/video then we have to rename to GUI again :-)

**Reply by cnervi** [reply]
then you could use just Terminal.UI :D

**Reply by BDisp** [reply]
No, because it will have a Graphical UI 😄

**Reply by tznind** [reply]
> No, because it will have a Graphical UI 😄

Indeed! And to elaborate on this point: one of the outstanding features we are looking to implement one day is support for directly embedding graphics (images) into the Terminal.  This is possible through the 'sixel' system which is supported by some (but not all) terminals.  For more info on sixel see here: https://iterm2.com/documentation-images.html and for the Terminal.Gui discussion see here: https://github.com/gui-cs/Terminal.Gui/discussions/1265

**Reply by tznind** [reply]
Also before we get into 'what is and isn't graphics' theres already a big discussion on that exact topic! in https://github.com/gui-cs/Terminal.Gui/discussions/1458

**Reply by cnervi** [reply]
as soon as you say graphics I though of sixel. for me at leasts text means character based, graphical means pixel or sprite based. and terminal.ui would be more neutral bc is not cli is a not specified UI, TUI framebuffer, sixel etc, but really you should just ignore me, I insist that I'm not insisting, lol

**Comment by tig** [comment]
You're not wrong about the name, but we're not renaming it. ;-)

**Reply by cnervi** [reply]
ty for answering, and yes is pretty late to renaming really
